### PR TITLE
dialects: (gpu) Adding `gpu.wait` op

### PR DESCRIPTION
**Pull request by tavakkoliamirmohammad**
This pull request adds the `gpu.wait` op for GPU asynchronous execution. The `gpu.wait` op generates a token that can be used to enable asynchronous execution combined with other operations in `gpu` dialect.

### CODE REVIEW

**Review by PapyChacal — COMMENTED** [review]
Nice, thanks!
Could tou simply add roundtrip (`tests/filecheck/dialects/gpu/ops.mlir`) and interoperability (`tests/filecheck/mlir-conversion/with-mlir/dialects/gpu/ops.mlir`) tests?

File: xdsl/dialects/gpu.py [hunk 1]
```diff
@@ -779,6 +795,7 @@ def verify_(self) -> None:
         SubgroupSizeOp,
         TerminatorOp,
         ThreadIdOp,
+        WaitOp,
         YieldOp,
     ],
     [
```
Comment (PapyChacal) [inline xdsl/dialects/gpu.py:801]:
```suggestion
    [
        AsyncTokenType
```
Oops, this is on me, I should have done it before!

**Review by PapyChacal — APPROVED** [review]
Perfect, thanks 😀

**Comment by tavakkoliamirmohammad** [comment]
Thank You! :)

**Comment by PapyChacal** [comment]
Feel free to merge when you want 🙂

**Comment by tavakkoliamirmohammad** [comment]
I don't have write access

**Comment by tavakkoliamirmohammad** [comment]
@PapyChacal Would you please commit these changes? (I don't have the write access)

**Pull request merged**